src/lib/message-router.js

```diff
@@ -105,18 +106,39 @@ MessageRouter.prototype._handleRequest = function * (request, sender) {
  * @param {Route[]} routes
  * @param {IlpAddress} sender
  */
-MessageRouter.prototype.receiveRoutes = function * (routes, sender) {
-  validate('Routes', routes)
-  let gotNewRoute = false
+MessageRouter.prototype.receiveRoutes = function * (payload, sender) {
+  validate('RoutingUpdate', payload)
+  debug('receiveRoutes sender:', sender)
+  let routes = payload.new_routes
+
+  let holdDownTime = payload.hold_down_time
+  this.routingTables.bumpConnector(sender, holdDownTime)
+  let potentiallyUnreachableLedgers = payload.unreachable_through_me
+  let lostLedgerLinks = []
+  if (potentiallyUnreachableLedgers.length > 0) {
+    log.info('informed of broken routes to:', potentiallyUnreachableLedgers, ' through:', sender)
+    for (const ledger of potentiallyUnreachableLedgers) {
+      lostLedgerLinks.push(...this.routingTables.invalidateConnectorsRoutesTo(sender, ledger))
+    }
+  }
 
+  if (!routes && (lostLedgerLinks.length === 0)) { // just a heartbeat
```

Comment:
When would `!routes` ever trigger? `payload` is a `RoutingUpdate`, which means `payload.new_routes` is always an `Array`.